### PR TITLE
Add competitive landscape for batch changes

### PR DESCRIPTION
**Pull request by malomarrec**
A short, narrative form competitive landscape for Batch Changes. It is meant to be succinct and convey the main storyline, rather than the details.
Inspired by pages such as https://about.gitlab.com/direction/release/pages/#competitive-landscape

### CODE REVIEW

File: content/product/product_areas.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+# Sourcegraph product areas
```
Comment (jasonyavorsky) [inline content/product/product_areas.md]: @malomarrec this file came in due to an issue with .gitignore I think, you should remove it from your PR.

File: content/company/strategy/code-graph/batch-changes/index.md [hunk 2]
```diff
@@ -46,6 +46,12 @@ To deliver our vision, we will:
 - Then, release Batch Changes on Sourcegraph Cloud, and test and learn how we can serve individuals and smaller teams.
 - Eventually, address the problem end to end by becoming the go-to place for code change tools and recipes.
 
+## Competitive landscape
```
Comment (jasonyavorsky) [inline content/company/strategy/code-graph/batch-changes/index.md:49]: This is really good, thanks for adding @malomarrec ! Thanks for summarizing and including in your strategy.

File: content/company/strategy/code-graph/batch-changes/index.md [hunk 3]
```diff
@@ -46,6 +46,12 @@ To deliver our vision, we will:
...
+
+We expect to see more vertical competition from companies focused on a language ecosystem, that enter the market by building refactoring frameworks for a given language ecosystem, then competing with Batch Changes on tooling to apply and track code changes. [Moderne](https://moderne.io/)/[openrewrite](https://github.com/openrewrite/rewrite) is the first significant example of this, focused on the Java ecosystem. We are committed to remaining language-agnostic, which includes using Batch Changes to apply and track changesets created with openrewrite or any other refactoring framework. We also think there is an opportunity in leveraging precise code intelligence to create radically better code change tools that support many languages in the future.
```
Comment (jasonyavorsky) [inline content/company/strategy/code-graph/batch-changes/index.md]: @macraig do you think there's an update on your side regarding this? Or is it really more that Batch Changes will be a customer of the features you drive, and there aren't integration considerations into how you're thinking about prioritization right now?

**Review by jasonyavorsky — APPROVED** [review]

File: content/company/strategy/code-graph/batch-changes/index.md (hunk 3 above)
Comment (macraig) [reply content/company/strategy/code-graph/batch-changes/index.md]: @jyavorska making it so other teams like Batch Changes can leverage our metadata is part of our mission statement. We haven't prioritized any specific work on that front, but it's something that we always keep in mind to make sure we're building for a future "Code intel platform" that will have Batch Changes, Search and Code Insights as our internal customers.

Comment (jasonyavorsky) [reply content/company/strategy/code-graph/batch-changes/index.md]: Sounds good, thanks Maria!

**Review by chrispine — APPROVED** [review]
Thanks for writing this up!

**Review by yinquanteo — APPROVED** [review]
Looks great!

**Pull request merged**